eduvpn/ui/stats.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import time
+from datetime import timedelta
+from gettext import gettext
+from pathlib import Path
+from functools import lru_cache
+
+from ..nm import get_iface, get_ipv4, get_ipv6, get_timestamp
+from ..utils import get_human_readable_bytes
+
+logger = logging.getLogger(__name__)
+
+LINUX_NET_FOLDER = Path("/sys/class/net")
+
+
+class NetworkStats:
+    def __init__(self):
+        self.default_text = gettext("N/A")
+
+    # These properties define an LRU cache
+    # This cache is used so that we do not query these every second
+    @property  # type: ignore
```

Comment:
why do these types need to be ignored?

Reply:
Because mypy doesn't really like cached properties, for that you need ```@functools.cached_property```, but that is only available for python 3.8 and up

Reply:
see https://github.com/python/mypy/issues/1362